From 4b59b12b1171359858e1712e1671488c491d3eab Mon Sep 17 00:00:00 2001
From: Ivan Shvedunov <ivan.shvedunov@travelping.com>
Date: Wed, 23 Feb 2022 16:18:34 +0300
Subject: [PATCH 07/12] Use heapsize 2G for UPG-VPP tests

---
 test/framework.py | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/test/framework.py b/test/framework.py
index c85dec5db..4c30c4940 100644
--- a/test/framework.py
+++ b/test/framework.py
@@ -458,11 +458,7 @@ class VppTestCase(CPUInterface, unittest.TestCase):
         cls.vpp_cmdline.extend(
             [
                 "}",
-                "physmem",
-                "{",
-                "max-size",
-                "32m",
-                "}",
+                "heapsize", "2G",
                 "statseg",
                 "{",
                 "socket-name",
-- 
2.32.0 (Apple Git-132)

