From f5d1cbe91840c9a872ff14f76fa6e4cf4e073487 Mon Sep 17 00:00:00 2001
From: Ivan Shvedunov <ivan.shvedunov@travelping.com>
Date: Wed, 23 Feb 2022 16:18:34 +0300
Subject: [PATCH] Use heapsize 2G for UPG-VPP tests

---
 test/asf/asfframework.py | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/test/asf/asfframework.py b/test/asf/asfframework.py
index 24880044c..aa77cc19c 100644
--- a/test/asf/asfframework.py
+++ b/test/asf/asfframework.py
@@ -417,11 +417,7 @@ class VppAsfTestCase(CPUInterface, unittest.TestCase):
         cls.vpp_cmdline.extend(
             [
                 "}",
-                "physmem",
-                "{",
-                "max-size",
-                "32m",
-                "}",
+                "heapsize", "2G",
                 "statseg",
                 "{",
                 "socket-name",
-- 
2.45.2

