From 8ef087923cfd3182cee0ee8d123220a7609721b8 Mon Sep 17 00:00:00 2001
From: Sergey Matov <sergey.matov@travelping.com>
Date: Fri, 14 Oct 2022 14:56:52 +0400
Subject: [PATCH] vlib: include dma.h into install header files

---
 src/vlib/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/vlib/CMakeLists.txt b/src/vlib/CMakeLists.txt
index 69bd18b6d..74069681d 100644
--- a/src/vlib/CMakeLists.txt
+++ b/src/vlib/CMakeLists.txt
@@ -149,6 +149,7 @@ add_vpp_library(vlib
   unix/unix.h
   vlib.h
   vmbus/vmbus.h
+  dma/dma.h
 
   API_FILES
   pci/pci_types.api
-- 
2.30.2

