From 059e086ce72c29b0fbf30ae06eb9b3381b720fe7 Mon Sep 17 00:00:00 2001
From: Ivan Shvedunov <ivan.shvedunov@travelping.com>
Date: Fri, 20 Oct 2023 04:01:30 +0400
Subject: [PATCH] test: fix import in vpp_ipsec.py

Type: fix
Signed-off-by: Ivan Shvedunov <ivan4th@gmail.com>
---
 test/vpp_ipsec.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/vpp_ipsec.py b/test/vpp_ipsec.py
index e354cfc8a..ac589935d 100644
--- a/test/vpp_ipsec.py
+++ b/test/vpp_ipsec.py
@@ -1,3 +1,4 @@
+import socket
 from vpp_object import VppObject
 from ipaddress import ip_address
 from vpp_papi import VppEnum
-- 
2.45.2

