From e9334e5f1b5a249e0a85adeaf8cd337b504ab8ab Mon Sep 17 00:00:00 2001
From: Ivan Shvedunov <ivan.shvedunov@travelping.com>
Date: Fri, 20 Oct 2023 04:01:30 +0400
Subject: [PATCH] test: fix import in vpp_ipsec.py

Type: fix
Signed-off-by: Ivan Shvedunov <ivan4th@gmail.com>
---
 test/vpp_ipsec.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/vpp_ipsec.py b/test/vpp_ipsec.py
index eb0209fc5..4030de6c1 100644
--- a/test/vpp_ipsec.py
+++ b/test/vpp_ipsec.py
@@ -1,3 +1,4 @@
+import socket
 from vpp_object import VppObject
 from ipaddress import ip_address
 from vpp_papi import VppEnum
-- 
2.40.0

