From e62404d4fd98a51fb79b8db3233a8bd7439aa33c Mon Sep 17 00:00:00 2001
From: Vladimir Zhigulin <vladimir.jigulin@travelping.com>
Date: Mon, 30 Jun 2025 15:23:55 +0200
Subject: [PATCH] fix: af_xdp: invalid recvmsg argument

---
 src/plugins/af_xdp/input.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/plugins/af_xdp/input.c b/src/plugins/af_xdp/input.c
index 0d8904196..93b7730c0 100644
--- a/src/plugins/af_xdp/input.c
+++ b/src/plugins/af_xdp/input.c
@@ -88,7 +88,14 @@ af_xdp_device_input_refill_db (vlib_main_t * vm,
 
   if (clib_spinlock_trylock_if_init (&rxq->syscall_lock))
     {
-      int ret = recvmsg (rxq->xsk_fd, 0, MSG_DONTWAIT);
+      struct msghdr msg = {0};
+      struct iovec iov = {0};
+      msg.msg_iov = &iov;
+      msg.msg_iovlen = 1;
+      iov.iov_len = 0;
+      iov.iov_base = NULL;
+
+      int ret = recvmsg (rxq->xsk_fd, &msg, MSG_DONTWAIT);
       clib_spinlock_unlock_if_init (&rxq->syscall_lock);
       if (PREDICT_FALSE (ret < 0))
 	{
-- 
2.49.0

