From ca6ca5aa5e2a150e7b1f76fed06b8d175fd4038a Mon Sep 17 00:00:00 2001
From: Vladimir Zhigulin <vladimir.jigulin@travelping.com>
Date: Thu, 3 Jul 2025 15:07:08 +0200
Subject: [PATCH] use no-omit-frame-pointer for release image

---
 src/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7c4f8db70..272ef1eb7 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -148,10 +148,10 @@ set(CMAKE_C_FLAGS_RELEASE "")
 set(CMAKE_C_FLAGS_DEBUG "")
 
 if (${CMAKE_BUILD_TYPE_LC} MATCHES "release")
-  add_compile_options(-O3 -fstack-protector -fno-common)
+  add_compile_options(-O3 -fstack-protector -fno-common -fno-omit-frame-pointer)
   add_compile_definitions(_FORTIFY_SOURCE=2)
 elseif (${CMAKE_BUILD_TYPE_LC} MATCHES "debug")
-  add_compile_options(-O0 -fstack-protector -fno-common)
+  add_compile_options(-O0 -fstack-protector -fno-common -fno-omit-frame-pointer)
   add_compile_definitions(CLIB_DEBUG)
 elseif (${CMAKE_BUILD_TYPE_LC} MATCHES "coverity")
   add_compile_options(-O0)
-- 
2.49.0

